### ISSUE TITLE
request.user.is_authenticated

### ISSUE THREAD

**Issue by chizcracker** [ozkatz/django-announce#2]
inside AnnounceCookieMiddleware.process_response(), I saw a line calling 

response.user.is_authenticated which calls 

from django.contrib.auth import get_user
request._cached_user = get_user(request)

-> 

104  -> def get_user(request):
105         from django.contrib.auth.models import AnonymousUser
106         try:
107             user_id = request.session[SESSION_KEY]
108             backend_path = request.session[BACKEND_SESSION_KEY]
109             backend = load_backend(backend_path)
110             user = backend.get_user(user_id) or AnonymousUser()

"backend" here is 'django.contrib.auth.backends.ModelBackend'.  Thus, "backend.get_user(user_id)" calls
->

 61  def get_user(self, user_id):
 62             try:
 63                 return User.objects.get(pk=user_id)
 64             except User.DoesNotExist:
 65                 return None

which seems like a blocking call to mysqldb.  

Is there a setting / backend that I am missing? I thought using node.js (which is async server) requires ( if we want to use async features..) non-blocking db calls...

Please tell me I'm wrong... I think this app is amazing except those lines, and I really want to use this in my project. 

Now I am thinking of writing a session sharing middleware (or just use django-redis-sessions) to let node.js read session info from redis.

**Comment by ozkatz** [comment]
Hi - the Announce library is constructed out of two parts: the node.js part which indeed uses nonblocking IO libraries, and the client part (in this case Django) which doesn't. This is because there is no need to async access here. 

Django itself uses blocking IO (and there is really nothing wrong with that), and django.contrib.auth uses normal DB calls to read and write from a database. Even if you were to write some redis backend for sessions, accessing redis is a blocking operation (unless you use gevent and monkey patch the socket library, but you could also do that with some of the "normal" DB drivers as well)..

In short - the node.js part of Announce is nonblocking on IO, and will remain so. The Django (or other) client, has no reason to be non-blocking. The whole idea behind Announce is to let node take care of sockets and asynchronicity, while your webapp can remain untouched. I hope this clarifies things for you.
